### PR TITLE
Implement landing page based on Issue #291 + Set up frontend testing

### PR DESCRIPTION
**Pull request by sibinhho99**
## Description

This PR implements new landing page for SSID based on https://github.com/WING-NUS/SSID/issues/291 .

It also sets up and include inaugural frontend tests using RSpec + Capybara +  Selenium.

## Related Issue

[https://github.com/WING-NUS/SSID/issues/291](https://github.com/WING-NUS/SSID/issues/291) 
[https://github.com/WING-NUS/SSID/issues/301](https://github.com/WING-NUS/SSID/issues/301)
[https://github.com/WING-NUS/SSID/issues/320](https://github.com/WING-NUS/SSID/issues/320)

## Motivation and Context

- Current landing page of SSID is outdated + basic
- Currently there's no frontend test, and @huyuxin0429 proposed writing tests together with feature

## How Has This Been Tested?

Local testing + add few automated tests for frontend using Capybara + RSpec + Selenium

## Screenshots (if appropriate):
See https://youtu.be/Tm4ZTjTYF2E for both landing page demo and Selenium test demo

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by huyuxin0429** [comment]
Thanks for pushing the changes in! I'll take some time to review the PR. I have not implemented the documentation for frontend testing yet, so feel free to document the relevant instructions.

**Comment by huyuxin0429** [comment]
Hmm the code looks alright, could you run eslint on this?

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -26,17 +26,15 @@ along with SSID.  If not, see <http://www.gnu.org/licenses/>.
...
-        <div class="site-login w-20">
-          <ul id="menu" <%= %Q{ class="admin"}.html_safe if user_signed_in? and current_user.is_admin %>>
-            <li <%= highlight_if_current new_user_session_path %>><%= link_to "Login", new_user_session_path %></li>
-            <li <%= highlight_if_current new_user_registration_path %>><%= link_to "Sign Up", new_user_registration_path%></li>
-          </ul>
-        </div>
+          <h3 id="site-banner-login"><a href='/' style="color:inherit; text-decoration:inherit">Student Submissions Integrity Diagnosis System</a></h3>
+          <div class="site-login w-20">
+            <ul id="menu" <%= %Q{ class="admin"}.html_safe if user_signed_in? and current_user.is_admin %>>
+              <li> <button type="button" style="background-color: transparent; color: #003d7c; height: 3rem; padding: 1rem" class="btn py-1" onclick="location.href='/login';">Log In</button> </li>
```
Comment (joshenx) [inline app/views/layouts/application.html.erb:32]: Should the inline styling be extracted into a css class?

Comment (huyuxin0429) [reply app/views/layouts/application.html.erb:32]: I'll merge this version in first, @joshenx you can do refactoring on this afte the merge

**Comment by huyuxin0429** [comment]
@sibinhho99 I have added chrome and chrome driver installation into the linter and tester scripts
Can you update the existing tests to run them headless mode?
I am thinking of something like this:
    Capybara.current_driver = :selenium_chrome_headless 
Additional information from here
https://github.com/actions/runner-images/issues/9

**Review by huyuxin0429 — CHANGES_REQUESTED** [review]
Fix failing test cases

**Comment by huyuxin0429** [comment]
Hmm thanks for the update, I'll mess around with the chromedriver related settings and see what I can do

**Pull request merged**